src/ds/ds-options.cpp

```diff
@@ -690,7 +690,22 @@ namespace librealsense
     void hdr_conditional_option::set(float value)
     {
         if (_hdr_cfg->is_config_in_process())
-            _hdr_option->set(value);
+        {
+            if( _hdr_cfg->is_enabled() )
+            {
+                // Changing exposure while HDR is enabled was requested by customers.It is currently disabled by D400
+                // FW, so we workaround it by disabling HDR,changing exposure and enabling again.Disabling/Enabling
+                // resets the sequence index so we need to keep and restore it.
+                auto _current_hdr_sequence_index = _hdr_cfg->get( RS2_OPTION_SEQUENCE_ID );
+                _hdr_cfg->set( RS2_OPTION_HDR_ENABLED, 0, { 0, 1, 1, 0 } );
+                _hdr_cfg->set( RS2_OPTION_SEQUENCE_ID, _current_hdr_sequence_index, { 0, 2, 1, 0 } );
+                _hdr_option->set( value );
+                _hdr_cfg->set( RS2_OPTION_HDR_ENABLED, 1, { 0, 1, 1, 0 } );
+                _hdr_cfg->set( RS2_OPTION_SEQUENCE_ID, _current_hdr_sequence_index, { 0, 2, 1, 0 } );
```

Comment:
Can you add a comment why this is done after the enabling? (so future changes wont change the order)

Reply:
The comment says `Disabling/Enabling resets the sequence index so we need to keep and restore it.`
Enabling HDR resets the the sequence so we need to restore it. For example if the sequence was 2 before changing the exposure, without restoring FW will start from 1 again.